packages/features/filters/lib/getTeamsFiltersFromQuery.ts

```diff
@@ -0,0 +1,28 @@
+import type { ParsedUrlQuery } from "querystring";
+import { z } from "zod";
+
+import { queryNumberArray } from "@calcom/lib/hooks/useTypedQuery";
+import type { RouterOutputs } from "@calcom/trpc/react";
+
+export type IEventTypesFilters = RouterOutputs["viewer"]["eventTypes"]["listWithTeam"];
+export type IEventTypeFilter = IEventTypesFilters[0];
+
+// Use filterQuerySchema when parsing filters out of query, so that additional query params(e.g. slug, appPages) aren't passed in filters
+export const filterQuerySchema = z.object({
+  teamIds: queryNumberArray.optional(),
+  userIds: queryNumberArray.optional(),
+});
+
+export const filterQuerySchemaStrict = z.object({
```

Comment:
strict ensures that userIds and teamIds are already transformed and is suitable to be used by trpc.